src/Sarif/Readers/FlagsEnumConverter.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif.Readers
+{
+    public class FlagsEnumConverter : JsonConverter
```

Comment:
Shouldn't be called this if in fact it only treats`SuppressionStates`. Did you intend to make it a generic type?